src/bpftrace.cpp

```diff
@@ -1503,6 +1551,30 @@ int BPFtrace::poll_perf_events()
   return ready;
 }
 
+int BPFtrace::poll_debug_output()
+{
+  auto events = std::vector<struct epoll_event>(1);
+  int ready = epoll_wait(debug_epollfd_, events.data(), 1, timeout_ms);
```

Comment:
Not sure if `timeout_ms` should be used here. It unnecessarily delays the event loop right? Is it possible to do no timeout?

Reply:
@danobi I originally planned to use a new thread to read event data independently, but this would complicate printing to standard output. So I switched to using the epoll, which avoids thread switching and may be less prone to problems.

I think the epoll_wait timeout is compulsory, otherwise it will block indefinitely until a debug event comes in.

Reply:
According to the man page:

> Specifying a timeout of -1
> causes epoll_wait() to block indefinitely, while specifying a timeout equal to zero cause epoll_wait()
> to return immediately, even if no events are available.

So maybe 0 will work

Reply:
Oops, how careless i am...